hail/src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -702,8 +702,25 @@ object Interpret {
             FastIndexedSeq(classInfo[Region], LongInfo, LongInfo), UnitInfo,
             extracted.seqPerElt)
 
-          val read = extracted.deserialize(ctx, spec, physicalAggs)
-          val write = extracted.serialize(ctx, spec, physicalAggs)
+          case class WrappedByteArray(var bytes: Array[Byte])
+
+          val read = {
+            val deserialize = extracted.deserialize(ctx, spec, physicalAggs)
+            (a: WrappedByteArray) => {
+              val r = Region(Region.SMALL)
+              val res = deserialize(r, a.bytes)
+              a.bytes = null
+              RegionValue(r, res)
+            }
+          }
+          val write = {
+            val serialize = extracted.serialize(ctx, spec, physicalAggs)
+            (rv: RegionValue) => {
+              val a = serialize(rv.region, rv.offset)
+              rv.region.clear()
```

Comment:
I think this (and the other `.clear()`s in this block) can be the slightly stronger `region.close()`. Not a huge difference, but this way you end up releasing the underlying RegionMemory for reuse before the GC runs. (And I tend to think of the difference as "I'm done with the value but I might want to reuse the region" vs. "I'm done with the region entirely", and I think the latter applies here.)

Reply:
Yup, that was a mistake.